OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs

```diff
@@ -30,29 +30,32 @@ protected override void OnFirstRun(Actor self)
 
 			if (!aircraft.Info.TakeOffOnResupply)
 			{
-				ChildActivity = ActivityUtils.SequenceActivities(
-					aircraft.GetResupplyActivities(host)
-					.Append(new AllowYieldingReservation(self))
-					.Append(new WaitFor(() => NextInQueue != null || aircraft.ReservedActor == null))
-					.ToArray());
+				ChildActivity = ActivityUtils.SequenceActivities(self, aircraft.GetResupplyActivities(host).ToArray());
+				QueueChild(self, new AllowYieldingReservation(self));
+				QueueChild(self, new WaitFor(() => NextInQueue != null || aircraft.ReservedActor == null));
 			}
 			else
 			{
-				ChildActivity = ActivityUtils.SequenceActivities(
-					aircraft.GetResupplyActivities(host)
-					.Append(new AllowYieldingReservation(self))
-					.Append(new TakeOff(self, (a, b, c) => NextInQueue == null && b.NextInQueue == null))
-					.ToArray());
+				ChildActivity = ActivityUtils.SequenceActivities(self, aircraft.GetResupplyActivities(host).ToArray());
+				QueueChild(self, new AllowYieldingReservation(self));
+				QueueChild(self, new TakeOff(self, (a, b, c) => NextInQueue == null && b.NextInQueue == null));
 			}
 		}
 
 		public override Activity Tick(Actor self)
 		{
+			if (ChildActivity != null)
+			{
+				ChildActivity = ActivityUtils.RunActivity(self, ChildActivity);
+				return this;
+			}
+
 			// Conditional fixes being able to stop aircraft from resupplying.
-			if (IsCanceled && NextInQueue == null)
-				OnFirstRun(self);
+			if (IsCanceling && NextInQueue != null)
+				return NextActivity;
 
-			return NextActivity;
+			OnFirstRun(self);
```

Comment:
Asking just in case: Are we really sure this won't run a 2nd time on the tick after the ChildActivity finishes?

Reply:
If we are, maybe you should leave at least leave a comment right above this line explaining why.

Reply:
No, this is a mistake. Fixed.